.github/workflows/tkg_integration_tests_clusterclass.yaml

```diff
@@ -149,30 +280,123 @@ jobs:
         env:
-          NUM: ${{ github.run_id }}
-          OCI_REGISTRY: ${{ steps.set_registry.outputs.registry }}/tanzu_framework/github-actions/${{ steps.extract_reg_dir.outputs.regdir }}
-          COMPONENTS: "pkg/v2/tkr/controller/tkr-source pkg/v2/tkr/controller/tkr-status pkg/v1/sdk/features addons pkg/v2/tkr/webhook/infra-machine pkg/v1/sdk/capabilities pkg/v2/tkr/webhook/tkr-conversion pkg/v2/tkr/webhook/cluster/tkr-resolver pinniped-components/tanzu-auth-controller-manager pkg/v2/object-propagation"
-          BUILD_VERSION: "v0.21.0"
-          IMG_VERSION_OVERRIDE: latest
+          _MANAGEMENT_PACKAGE_REPO_IMAGE: ${{ steps.set_registry.outputs.registry }}/tanzu_framework/github-actions/${{ steps.extract_reg_dir.outputs.regdir }}/management:v0.21.0
+          _MANAGEMENT_PACKAGE_VERSION: "0.21.0"
+          TKG_CUSTOM_IMAGE_REPOSITORY: "projects-stg.registry.vmware.com/tkg"
+
+
+  integ-tests-aws-cc:
+    name: TKG Integration Tests CAPA (ClusterClass)
+    runs-on: ubuntu-latest
+    needs: build-and-publish-packages
+    env:
+      aws_client_id: ${{secrets.AWS_ACCESS_KEY_ID}}
+      gcp_bucket_sa: ${{ secrets.GCP_BUCKET_SA }}
+    steps:
+      - name: Free disk space and clean up installed plugins on runner
         run: |
-          make local-registry
-          make docker-build
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /home/tanzu/.cache
+          sudo rm -rf /home/tanzu/.kube
+          sudo rm -rf /home/tanzu/.local/share/tanzu-cli/*
+          sudo sh -c 'find /tmp -type d -name "go-build*" -mtime +3 | xargs rm -rf'
 
-      - name: Publish management packages
-        env:
-          NUM: ${{ github.run_number }}
-          OCI_REGISTRY: ${{ steps.set_registry.outputs.registry }}/tanzu_framework/github-actions/${{ steps.extract_reg_dir.outputs.regdir }}
-          COMPONENTS: "pkg/v2/tkr/controller/tkr-source pkg/v2/tkr/controller/tkr-status pkg/v1/sdk/features addons pkg/v2/tkr/webhook/infra-machine pkg/v1/sdk/capabilities pkg/v2/tkr/webhook/tkr-conversion pkg/v2/tkr/webhook/cluster/tkr-resolver pinniped-components/tanzu-auth-controller-manager pkg/v2/object-propagation"
-          BUILD_VERSION: "v0.21.0"
-          IMG_VERSION_OVERRIDE: latest
+      - name: show environment
+        uses: hmarr/debug-action@v2
+
+      - name: Extract PR Number
+        uses: Dovyski/payload-info-action@master
+        id: get_pr_num
+        with:
+          # when event is a pull request, obtaining the PR number is obvious
+          filter_pull_request: '.number'
+          # when event is a push (merge of PR), since we require linear history,
+          # we are not even generating a merge commit that can help identify the
+          # PR number, so don't even try. Instead we just hard-code to a dummy
+          # value.
+          filter_push: 1009
+
+      - name: Determine registry directory
```

Comment:
I think all the steps in the integ-tests-xxx that involves figuring out the registry related variables can also be removed. Instead we should be able to reference the variables computed in the build-and-publish-packages job instead. See if something like jobs._jobid_.steps.get_registry_dir.outputs.value works. (You will want to set a friendly id: value for each of the jobs, but especially the first one)

Reply:
This was a great suggestion. It needed something like `needs.<jobID>.outputs.<outputID>`